### DISCUSSION TITLE
Support for typographic correct quotation marks

### DISCUSSION THREAD

**Discussion by mrwunderbar666** [Witiko/markdown#190] [Q&A] [answered]
Thanks for making this package!

I was trying it out and noticed that quotation marks are not rendered correctly.

When I input this:

```markdown
This is in "quotation marks".
```

The output is:
```
This is in ”quotation marks”.
```

But my expected output would be:

```
This is in “quotation marks”.
```

I think it would be great if the parser could detect tokens that are within quotation marks and automatically wrap them with `\enquote{}` from the [csquotes](https://ctan.org/pkg/csquotes) package. This would allow further customization such as british style quotation marks, etc.

See a minimum working example here:

```tex

\documentclass{article}          
\usepackage{markdown}
\markdownSetup{pipeTables,tableCaptions,definitionLists,fencedCode,hybrid}

\begin{document}


\begin{markdown}

# Chapter 1

There was nothing so very remarkable in that; nor did Alice think it so **very** much out 
of the way to hear the rabbit say to itself, 
"dear, dear! I shall be too late!" (when she thought it over afterwards, 
it occurred to her that she ought to have wondered at this, 
but at the time it all seemed quite natural); but when 
the rabbit actually *took a watch out of its waistcoat-pocket*, 
and looked at it, and then hurried on, 
Alice started to her feet, for it flashed across her mind that 
she had never before seen a rabbit with either a waistcoat-pocket 
or a watch to take out of it, and, full of curiosity, she ran across the 
field after it, and was just in time to see it pop down 
a large rabbit-hole under the hedge. In another moment down 
went Alice after it, never once considering how in the world 
she was to get out again.

\end{markdown}
\end{document}
```

**Comment by Witiko** [comment, marked as answer]
@mrwunderbar666 Thank you for the proposal. The Markdown package could definitely support quotes as a syntax extension; for example, we already support [the `smartEllipses` option][1] for ellipses. However, in LaTeX, you can just as well use a package such as `csquotes`, which will achieve the same result for you: ([source][2])

``` tex
\usepackage{csquotes}
\MakeOuterQuote{"}
```

A good programmer is a lazy programmer. If we can leverage existing LaTeX code, perhaps extending the parser is just busywork.

 [1]: https://witiko.github.io/markdown/#option-smartellipses
 [2]: https://tex.stackexchange.com/a/446281/70941

**Reply by mrwunderbar666** [reply]
I didn't know that csquotes could do that now. When I started using that years ago it couldn't do that, how time flies.

Thanks for that!
